File: ruby/4.rb
# A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

# Find the largest palindrome made from the product of two 3-digit numbers.

def largest_palindrome
  product_array = []
  999.downto(100).each do |i|
    999.downto(100).each do |j|
      product = i*j
      if check(product)==true
        product_array << product
      end
    end
  end
  product_array.max
end

def check(num)
  if num.to_s===num.to_s.reverse
    return true
  end
end

largest_palindrome

# 906609

File: swift/4.swift
// #4
// Find the largest palindrome made from the product of two 3-digit numbers

import Cocoa

func check(num: Int) -> Bool {
    var number_string = String(num);
    var number_array: [Character] = []
    for character in number_string {
        number_array.append(character)
    }
    
    return number_array == number_array.reverse() ? true : false
}

func largest_palindrome() -> Int {
    var array: [Int] = []
    for var i = 100; i < 999; i++ {
        println(i)
        for var j = 100; j < 999; j++ {
            var product = i*j;
            if (check(product)==true) {
                println(j)
                array.append(product)
                break
            }
        }
    }
    return array.reduce(Int.min, {max($0, $1)})
}

largest_palindrome()

// 906609
